src/v/cluster/rm_stm.cc

```diff
@@ -597,6 +597,11 @@ rm_stm::replicate(
     }
 }
 
+ss::future<> rm_stm::stop() {
+    return raft::state_machine::stop().finally(
+      [this] { auto_abort_timer.cancel(); });
```

Comment:
should we cancel the timer _before_ waiting on background work to stop? otherwise it seems as though an active timer might start more background work (and throw a gate closed exception) after stop() completes but before the finally() block runs.

Reply:
I'm fine either way, we execute gate's futures as `(void)ss::with_gate(_gate` so the exception will be ignored, moving the cancellation in front of `raft::state_machine::stop()`

Reply:
`(void)ss::with_gate(_gate` will not ignore the exception. the seastar reactor will complain loudly that an exceptional future was ignored.

Reply:
good point, I've moved the cancellation in front of `raft::state_machine::stop()`